proxy/api/src/patch.rs

```diff
@@ -0,0 +1,144 @@
+//! [`list`] all the [`Patch`]es for project.
+
+use either::Either;
+use radicle_git_ext::Oid;
+use radicle_source::surf::git::RefScope;
+use serde::{Deserialize, Serialize};
+
+use radicle_daemon::{net, project, signer::BoxedSigner, state, PeerId, Urn};
+
+use crate::identity;
+
+const TAG_PREFIX: &str = "radicle-patch/";
+
+/// A patch is a change set that a user wants the maintainer to merge into a projects default
+/// branch.
+///
+/// A patch is represented by an annotated tag, prefixed with `radicle-patch/`.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Patch {
```

Comment:
Take this suggestion with a grain of salt :) There seems to be a mixture of data here. There's the `Patch` and where it came from. On top of that, `Identity` contains the `PeerId` so you could abstract over where it came from using an enum. So maybe something like this:

```rust
pub struct Patch {
  pub id: String,
  pub message: Option<String>,
  pub commit: Oid,
  pub merge_base: Option<Oid>,
}

pub enum Origin {
  Peer(PeerId),
  Identity(identity::Identity),
}

pub struct UserPatch {
  pub patch: Patch,
  pub origin: Origin,
}
```

Reply:
Alternatively, how do you feel about replacing `peer_id` and `identity` with
```rust
peer: project::Peer<peer::Status<Person>>
```

Reply:
I thought about this but wasn't sure if it was the correct semantics. Not all variants of a `Status` have a `PeerId`, however, maybe it doesn't make sense for a `Patch` to exist if it's a non-replicated peer?

Reply:
> Not all variants of a `Status` have a `PeerId`,

But the outer `project::Peer` always has a peer ID so this works. I’ve updated the PR.

Reply:
Nice, looks good!